### ISSUE TITLE
Kuma and http proxy servers monitor

### ISSUE THREAD

**Issue by franciscopaniskaseker** [kumahq/kuma#4069]
### Description

**What I want:**
I want to monitor some proxy nodes that are just proxy to the real http server. Those nodes listen the same domain, but in different IPs.

In the ideal world we usually use Load Balance and we just need to monitor the LB. But due costs restrictions, sometimes people use DNS round robin and manually remove the DNS when one proxy is down, or they use a script to do that.

**What we can do now:**
- We can use ping, but it will not test the http service;
- We can use tcp port, but it will not test SSL and just check if 80 port is replying is not enough to catch http errors

**Possible solution:**
As we need to test the same domain in different iPs, an extra field to force the DNS resolution will fix the problem.

**Comment by lahabana** [comment]
Can you describe a little bit the role that Kuma plays here? Maybe define a little bit your setup and if running in k8s show some example manifests.

**Comment by franciscopaniskaseker** [comment]
Suppose this:
- one backend server
- two nginx proxy forwarding the traffic to backend server
- both nginx proxy are configured in the same domain (so the domain will do round robin to distribute users)

I want to monitor the nginx proxy nodes using http protocol (to get http codes). I can not setup kuma to monitor the domain, because the requests will go under round robin algorithm (the monitor will not represent well both servers). I need to setup two monitors (as I have two nginx proxy). Both will have same DOMAIN, but different IP.

With current kuma features, I can just test 80 and 443 ports of the IPs, but these tests can not guarantee that all nginx proxy nodes are working correctly. I can just guarantee the all tcp ports can reply.

**Comment by lahabana** [comment]
I'm guessing that the backend service, nginx proxies and clients to the nginx proxies are all running inside the mesh. Please correct me if that isn't the case.

If you use: [Kuma's healthcheck policies](https://kuma.io/docs/1.5.x/policies/health-check/) envoy will send health checks to each instance independently and remove the unhealthy ones from the endpoint list (beware of [envoy's panic threshold](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/panic_threshold) which can be counter intuitive when there's only a few instances).

**Comment by lahabana** [comment]
@franciscopaniskaseker does this answer your question?

**Comment by franciscopaniskaseker** [comment]
Sorry by big delay, I forgot to reply. I did not find how to configure healthy policy under web dashboard. However i do not think is the case (but I can be wrong).

Imagine these scenarios:

(1) normal scenario
```bash
user <-> nginx (xyz) <-> services
```
(2) special scenario, with proxies
```bash
          / <->  nginx proxy (1) <-> \
user - <-> nginx proxy (2) ------ <-> nginx (xyz) <-> services
          \ <->  nginx proxy (3) <-> /
```

I want to monitor nginx proxy (1, 2 and 3) and also the nginx (xyz) of the scenario 2. But I want http monitor, to check ssl problems and they return code.

**The problem:**
All 4 nginx services have different IPs, but they answer same virtualhost (suppose domain.com).
So, for example:
nginx proxy 1: 1.1.1.1 / domain.com
nginx proxy 2: 2.2.2.2 / domain.com
nginx proxy 2: 3.3.3.3 / domain.com
nginx xyz: 8.8.8.8 / domain.com

How can I check them using kuma? Today I can open the tcp port and test if the port is working, but will be better to check the http access to check the return code, the ssl etc.

So my idea is make possible to force domain.com being solved by specific IP. I can not do that through /etc/hosts, because kuma will eventually (will depend of the dns round robin give the problematic server ip) notify the server with problem and I will not know which server is the problem (because kuma is only aware about domain, and not the specific IP).

So, basically, i have 4 nginx servers listening same domain, but in different IPs, to check http return codes. All of them must work for users (sometimes the proxies are disabled, sometimes enable).

**Comment by michaelbeaumont** [comment]
@franciscopaniskaseker I'm going to close this issue, please don't hesitate to post here if you have more information!
